micro-rdk/src/common/button.rs

```diff
@@ -0,0 +1,111 @@
+use super::{generic::DoCommand, status::Status};
+
+use std::sync::{Arc, Mutex};
+use thiserror::Error;
+#[cfg(feature = "builtin-components")]
+use {
+    crate::{
+        common::{
+            config::ConfigType,
+            registry::{ComponentRegistry, Dependency},
+            status::StatusError,
+        },
+        google,
+    },
+    std::collections::HashMap,
+};
+
+pub static COMPONENT_NAME: &str = "button";
+
+pub type ButtonType = Arc<Mutex<dyn Button>>;
+
+#[derive(Debug, Error)]
+pub enum ButtonError {
+    #[error("failed to press button")]
+    FailedPress,
+    #[error("test error")]
+    TestError,
+    #[error(transparent)]
+    Other(#[from] Box<dyn std::error::Error + Send + Sync>),
```

Comment:
catch all error are useful for wrapping errors only defined in a target (like EspError) do we expect button to be build using esp (or other) specific implementation?

Reply:
maybe? what's the alternative for a module writer? should this just be a string instead?

what errors do you suggest we have?

Reply:
i guess we don't have a good idea of what failure mode of a button looks like or even how we would use it so let's leave it as is